src/test/regress/spec/isolation_logical_replication_nonsu_nonbypassrls.spec

```diff
@@ -0,0 +1,133 @@
+// isolation_logical_replication_nonsu_nonbypassrls
+// test moving a single shard that has rls
+// owned by a user that is neither superuser nor bypassrls
+// PG15 added extra permission checks within logical replication
+// this test makes sure that target table owners should still
+// be able to replicate despite RLS policies.
+// Relevant PG commit: a2ab9c06ea15fbcb2bfde570986a06b37f52bcca
+
+setup
+{
+    SET citus.max_cached_conns_per_worker to 0;
+    SET citus.next_shard_id TO 1234000;
+	SET citus.shard_count TO 4;
+	SET citus.shard_replication_factor TO 1;
+
+    CREATE USER new_user;
+    GRANT ALL ON SCHEMA public TO new_user;
+
+    CREATE TABLE dist(column1 int PRIMARY KEY, column2 int);
+    SELECT create_distributed_table('dist', 'column1');
+
+	SELECT get_shard_id_for_distribution_column('dist', 23) INTO selected_shard;
```

Comment:
Instead of this I started using `citus.shard_count = 1` when testing shard moves. I think it makes the tests a lot simpler

Reply:
I always prefer a shard count of 4. It's a usual Citus cluster, with shards in both workers. Also, each worker has more than 1 shard. I see it as a minimum Citus functioning scenario :D and I prefer to always run my tests with this setup, to be safe and to make sure we don't miss anything.